core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -712,6 +714,9 @@ private void parseSource(SearchContext context, SearchSourceBuilder source) thro
         if (source.minScore() != null) {
             context.minimumScore(source.minScore());
         }
+        if (source.profile() != null && source.profile()) {
```

Comment:
In this case, just following the existing convention in `SearchService` ... e.g. not setting a default value in `SearchSourceBuilder` and then using null to check.

But I personally think that's gross so I'd be happy to default `profile = false` and check explicitly.